tests/acceptance/dex/idtokenprovider.go

```diff
@@ -44,48 +50,55 @@ func (p *dexIdTokenProvider) implicitFlow() (map[string]string, error) {
 	if err != nil {
 		return nil, err
 	}
-	if authorizeResp.StatusCode < 300 || authorizeResp.StatusCode > 399 {
-		return nil, fmt.Errorf("Authorize - response error: '%s' - %s", authorizeResp.Status, readRespBody(authorizeResp))
+	if authorizeResp.StatusCode < 200 || authorizeResp.StatusCode > 399 {
+		return nil, errors.New(fmt.Sprintf("Authorize - response error: '%s' - %s", authorizeResp.Status, readRespBody(authorizeResp)))
 	}
 
 	// /auth/local?req=qruhpy2cqjvv4hcrbuu44mf4v
-	loginEndpoint := authorizeResp.Header.Get("location")
-	if strings.Contains(loginEndpoint, "#.*error") {
-		return nil, fmt.Errorf("Login - Redirected with error: '%s'", loginEndpoint)
+	var loginEndpoint string
+	if authorizeResp.StatusCode == 200 {
+		loginEndpoint, err = getLocalAuthEndpoint(authorizeResp.Body)
+		if err != nil {
+			return nil, errors.Wrapf(err, "while fetching link to static authentication")
+		}
+	} else {
```

Comment:
why u assuming that it is a redirection call? why we are not checking the status code?

Reply:
Added following condition:
```
	switch authorizeResp.StatusCode {
	case http.StatusFound:
	case http.StatusOK:
	default:
		return nil, fmt.Errorf("got unexpected response on authorize: %d - %s", authorizeResp.StatusCode, readRespBody(authorizeResp))
	}
```